acceptance/gossip_peerings_test.go

```diff
@@ -99,5 +123,42 @@ func TestGossipPeerings(t *testing.T) {
 	if err := l.Nodes[pickedNode].Restart(5); err != nil {
 		t.Fatal(err)
 	}
-	checkGossipPeerings(t, l, 20*time.Second)
+	checkGossip(t, l, 20*time.Second, hasPeers(len(l.Nodes)))
+}
+
+// TestGossipRestart verifies that the gossip network can be
+// re-bootstrapped after a time when all nodes were down
+// simultaneously.
+func TestGossipRestart(t *testing.T) {
+	l := localcluster.Create(*numNodes, stopper)
+	l.Start()
+	defer l.Stop()
+
+	log.Infof("waiting for initial gossip connections")
+	checkGossip(t, l, 20*time.Second, hasPeers(len(l.Nodes)))
+	checkGossip(t, l, time.Second, hasClusterID)
+	checkGossip(t, l, time.Second, hasSentinel)
+
+	// The replication of the first range is important: as long as the
+	// first range only exists on one node, that node can trivially
+	// acquire the leader lease. Once the range is replicated, however,
+	// nodes must be able to discover each other over gossip before the
+	// lease can be acquired.
+	log.Infof("waiting for range replication")
+	checkRangeReplication(t, l, 10*time.Second)
+
+	log.Infof("stopping all nodes")
+	for _, node := range l.Nodes {
+		node.Stop(5)
+	}
+
+	log.Infof("restarting all nodes")
+	for _, node := range l.Nodes {
+		node.Restart(5)
```

Comment:
Yeah, Start() takes some extra arguments; Restart() just reuses the configuration that was originally passed to Start().